rfcs/0064-documentation-format.md

```diff
@@ -0,0 +1,173 @@
+
+[reStructuredText (reST)](https://en.wikipedia.org/wiki/ReStructuredText) is a file
+format originally developed as part of the Docutils project for documenting the Python language.
+Since then, support was added for reST to Sphinx, a popular tool for documenting (Python) projects, and pandoc.
+
+With Sphinx it is possible to document various languages using the concept of [domains](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html). E.g., if we were to have a format for documenting Nix functions, we could implement a domain in Sphinx, as well as a parser that could parse Nix functions from comments and convert them to the Sphinx domain, as is done currently with the [Nixpkgs library](https://github.com/NixOS/nixpkgs/pull/53055).
+
+Language:
+- [Specification](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html)
+- [Primer](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html)
+- [Demo](https://docutils.readthedocs.io/en/sphinx-docs/user/rst/demo.html)
+
+Tooling:
+- [Sphinx](https://www.sphinx-doc.org/)
+- [Docutils](https://docutils.sourceforge.io/)
+- [Pandoc](https://pandoc.org/) can convert from/to reST to/from many other formats
+
+Examples of users:
+- Python
+- Linux kernel
+- CMake
+- Majority of Python packages
+
+### Asciidoc
+
+TODO: Short overview
+
+[Demo](https://github.com/opendevise/asciidoc-samples/blob/master/demo.adoc)
+
+Tooling:
+- [Antora](https://antora.org/)
+- [Asciidoctor](https://asciidoctor.org/)
+
+### Texinfo
+
+TODO: Short overview
+
+Powerful, interactive and very nice to use (check out `pinfo`), but harder to write.
+
+### Nix EDSL
+
+With a Nix EDSL, linking to options can become trivial and very natural. Users won't have to learn another language either. Docs could also be written directly next to the thing they document with some convenience functions for annotating values with docs.
+
+### Docbook
+
+TODO: Short overview
+
+[Primer](https://docbook.rocks/)
+
+### Comparisons
+
+| Format | Rendered in GitHub | Adoption | Standardized | Goal |
+| --- | --- | --- | --- | --- |
+| Markdown | Yes | Great among websites | No | Easy to use |
+| reStructuredText | Yes | Great among tech docs | Yes | Easy to use, customizable |
+| Asciidoc | Yes | ? | Yes | Easy to use, customizable |
+| Docbook | No | ? | Yes | Semantic structure |
+
+Cheatsheet comparison: http://hyperpolyglot.org/lightweight-markup
+
+- Linux kernel, why Sphinx/reStructuredText (2016): https://lwn.net/Articles/692704/
+- Why not Markdown: https://mister-gold.pro/posts/en/asciidoc-vs-markdown/
+
+TODO: More online comparisons?
+
+### Comparison of tools
+
+For the following comparison NixOS 19.09 is used.
+
+| Name         | Attribute             | Closure size |
+|--------------|-----------------------|--------------|
+| Sphinx       | `python3.pkgs.sphinx` | 195 MB       |
+| Pandoc       | `pandoc`              | 2.4 GB       |
+| Asciidoctor  | `asciidoctor`         | 1.0 GB       |
```

Comment:
Asciidoctor is now about 80% smaller on master, with further reductions to come.